h2quic/server_test.go

```diff
@@ -325,6 +328,58 @@ var _ = Describe("H2 server", func() {
 		Eventually(func() bool { return handlerCalled }).Should(BeTrue())
 	})
 
+	Context("HTTP/2 SETTINGS", func() {
+		var (
+			h2framer     *http2.Framer
+			hpackDecoder *hpack.Decoder
+			headerStream *mockStream
+		)
+
+		BeforeEach(func() {
+			headerStream = newMockStream(3)
+			hpackDecoder = hpack.NewDecoder(4096, nil)
+			h2framer = http2.NewFramer(nil, headerStream)
+		})
+
+		It("Handles a SETTINGS frame", func() {
+			headerStream.dataToRead.Write([]byte{
+				// length	  |type|flags| data stream id   |
+				0x0, 0x0, 0x0, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0,
+			})
+			err := s.handleRequest(session, headerStream, &sync.Mutex{}, hpackDecoder, h2framer)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("Disables pushes if requested", func() {
+			headerStream.dataToRead.Write([]byte{
+				// length	  |type|flags| data stream id   |
+				0x0, 0x0, 0x6, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0,
+				// SETTINGS identifier			   | Value
+				0x0, uint8(http2.SettingEnablePush), 0x0, 0x0, 0x0, 0x0,
+			})
+			err := s.handleRequest(session, headerStream, &sync.Mutex{}, hpackDecoder, h2framer)
+			Expect(err).ToNot(HaveOccurred())
+			pushEnabled, ok := s.pushEnabled[session.ConnectionID()]
```

Comment:
There's a short form: `Expect(s.pushEnabled).To(HaveKeyWithValue(session.ConnectionID(), true))`